product-service/handlers/get-product-by-id/index.ts

```diff
@@ -0,0 +1,29 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import products from '../../mocks/products';
+
+export const handler: APIGatewayProxyHandler = async (event, _context) => {
+  try {
+    const { productId } = event.pathParameters;
+    if (!productId) {
+      throw new Error('Please provide product id!');
+    }
+
+    const product = products.find(item => item.id === productId);
+    if (!product) {
+      throw new Error(`Product with id='${productId}' not found!`);
+    }
+
+    return {
+      statusCode: 200,
+      body: JSON.stringify(product),
+    };
+
+  }catch (error) {
+    console.error(error);
+    return {
+      statusCode: 400,
```

Comment:
I'd suggest implementing an errorHandler function, which would process varoius types of error. Currently, all the errors thrown will send the `400` statusCode, which is not correct (for example, if a product is not found, the `statusCode` should be `404`).

It's a wish for the future, no need to implement it right now. It won't affect your score.